src/lib/containers/table/datasets/DatasetItemsEditor.tsx

```diff
@@ -0,0 +1,478 @@
+      }
+    })
+    clearSelectedIds()
+  }
+
+  function removeSelectedItemsFromDataset() {
+    setDatasetToUpdate(dataset => ({
+      ...dataset!,
+      items: dataset!.items.filter(
+        datasetItem => !selectedIds.has(datasetItem.entityId),
+      ),
+    }))
+
+    clearSelectedIds()
+  }
+
+  function changeVersionOnItem(entityId: string, newVersion: number) {
+    setDatasetToUpdate(dataset => ({
+      ...dataset!,
+      items: dataset!.items.map(datasetItem =>
+        datasetItem.entityId === entityId
+          ? { entityId: entityId, versionNumber: newVersion }
+          : datasetItem,
+      ),
+    }))
+  }
+
+  const DatasetItemVersionRenderer = (
+    props: CellRendererProps<DatasetItemsEditorTableData>,
+  ) => {
+    return (
+      <DatasetEditorVersionRenderer
+        {...props}
+        toggleSelection={datasetItem => {
+          changeVersionOnItem(datasetItem.entityId, datasetItem.versionNumber)
+        }}
+      />
+    )
+  }
+
+  type SelectAllCheckboxRendererProps = {
+    datasetToUpdate: Dataset
+    selectedIds: Omit<Set<string>, 'add' | 'delete' | 'clear'>
+    addSelectedId: (...items: string[]) => void
+    clearSelectedIds: () => void
+    allItemsAreSelected: boolean
+  }
+  const SelectAllCheckboxRenderer = (props: SelectAllCheckboxRendererProps) => {
+    const { datasetToUpdate, clearSelectedIds, addSelectedId } = props
+    useTraceUpdate(props)
+    const isChecked = allItemsAreSelected
+
+    return datasetToUpdate ? (
+      <div
+        data-testid="Select All"
+        style={{ cursor: 'pointer' }}
+        onClick={() => {
+          if (isChecked) {
+            clearSelectedIds()
+          } else if (datasetToUpdate) {
+            addSelectedId(...datasetToUpdate.items.map(item => item.entityId))
+          }
+        }}
+      >
+        <Checkbox
+          label=""
+          className="SRC-pointer-events-none"
+          checked={isChecked}
+          disabled={datasetToUpdate.items.length === 0}
+          onChange={() => {
+            // no-op
+          }}
+        />
+      </div>
+    ) : (
+      <></>
+    )
+  }
+
+  const renderedSelectAllCheckbox = datasetToUpdate ? (
+    <SelectAllCheckboxRenderer
+      datasetToUpdate={datasetToUpdate}
+      selectedIds={selectedIds}
+      clearSelectedIds={clearSelectedIds}
+      addSelectedId={addSelectedId}
+      allItemsAreSelected={allItemsAreSelected}
+    />
+  ) : (
+    <></>
+  )
+
+  const defaultColumns: ColumnShape<DatasetItemsEditorTableData>[] = [
+    {
+      key: 'errorState',
+      width: 30,
+      cellRenderer: EntityErrorRenderer,
+    },
+    {
+      key: 'isSelected',
+      width: 40,
+      dataKey: 'isSelected',
+      headerRenderer: renderedSelectAllCheckbox,
+      cellRenderer: CheckboxRenderer,
+    },
+    {
+      key: 'name',
+      width: 350,
+      dataKey: 'entityId',
+      title: 'Name',
+      resizable: true,
+      style: { textOverflow: 'ellipsis', whiteSpace: 'nowrap' },
```

Comment:
ok to apply style to all EntityNameWithIconRenderers (specify css class in the renderer)?